include/inexor/vulkan-renderer/world/collision_query.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/world/collision.hpp"
+#include "inexor/vulkan-renderer/world/cube.hpp"
```

Comment:
I think you should be able to forward declare `Cube` if you add the explicit instantiations for `RayCubeCollision`.

Reply:
Done.